tests/Resources/util.test.js

```diff
@@ -0,0 +1,1551 @@
+			);
+		});
+
+		it('throws TypeError if super constructor has no prototype', () => {
+			const BaseClass = Object.create(null, {});
+
+			function MyStream() {
+				BaseClass.call(this);
+			}
+
+			should.throws(() => util.inherits(MyStream, BaseClass),
+				TypeError
+			);
+		});
+	});
+
+	describe('#isArray', () => {
+		it('should return true only if the given object is an Array', () => {
+			should.strictEqual(util.isArray([]), true);
+			should.strictEqual(util.isArray(Array()), true);
+			should.strictEqual(util.isArray(new Array()), true);
+			should.strictEqual(util.isArray(new Array(5)), true);
+			should.strictEqual(util.isArray(new Array('with', 'some', 'entries')), true);
+			should.strictEqual(util.isArray({}), false);
+			should.strictEqual(util.isArray({ push: function () {} }), false);
+			should.strictEqual(util.isArray(/regexp/), false);
+			should.strictEqual(util.isArray(new Error()), false);
+			should.strictEqual(util.isArray(Object.create(Array.prototype)), false);
+		});
+	});
+
+	describe('#isRegExp', () => {
+		it('should return true only if the given object is a RegExp', () => {
+			should.strictEqual(util.isRegExp(/regexp/), true);
+			should.strictEqual(util.isRegExp(RegExp(), 'foo'), true);
+			should.strictEqual(util.isRegExp(new RegExp()), true);
+			should.strictEqual(util.isRegExp({}), false);
+			should.strictEqual(util.isRegExp([]), false);
+			should.strictEqual(util.isRegExp(new Date()), false);
+			should.strictEqual(util.isRegExp(Object.create(RegExp.prototype)), false);
+		});
+	});
+
+	describe('#isDate', () => {
+		it('should return true only if the given object is a Date', () => {
+			should.strictEqual(util.isDate(new Date()), true);
+			should.strictEqual(util.isDate(new Date(0), 'foo'), true);
+			should.strictEqual(util.isDate(Date()), false);
+			should.strictEqual(util.isDate({}), false);
+			should.strictEqual(util.isDate([]), false);
+			should.strictEqual(util.isDate(new Error()), false);
+			should.strictEqual(util.isDate(Object.create(Date.prototype)), false);
+		});
+	});
+
+	describe('#isError', () => {
+		it('should return true only if the given object is an Error', () => {
+			should.strictEqual(util.isError(new Error()), true);
+			should.strictEqual(util.isError(new TypeError()), true);
+			should.strictEqual(util.isError(new SyntaxError()), true);
+			should.strictEqual(util.isError({}), false);
+			should.strictEqual(util.isError({ name: 'Error', message: '' }), false);
+			should.strictEqual(util.isError([]), false);
+			should.strictEqual(util.isError(Object.create(Error.prototype)), true);
+		});
+	});
+
+	describe('#isObject', () => {
+		it('should return true only if the given object is an Object', () => {
+			should.strictEqual(util.isObject({}), true);
+			should.strictEqual(util.isObject([]), true);
+			should.strictEqual(util.isObject(new Number(3)), true);
+			should.strictEqual(util.isObject(Number(4)), false);
+			should.strictEqual(util.isObject(1), false);
```

Comment:
<!--
  0 failure: 
  1 warning:  tests/Resources/u...
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/util.test.js;
  Line: 894;
-->

- :warning: tests/Resources/util.test.js line 894 – 'util.isObject' was deprecated since v4.0.0. (node/no-deprecated-api)